ray-curator/ray_curator/stages/audio/io/asr_inference.py

```diff
@@ -0,0 +1,140 @@
+from dataclasses import dataclass, field
+
+import nemo.collections.asr as nemo_asr
+
+from ray_curator.stages.base import CompositeStage, ProcessingStage
+from ray_curator.stages.io.reader.file_partitioning import FilePartitioningStage
+from ray_curator.tasks import SpeechEntry, _EmptyTask
+from ray_curator.tasks.file_group import FileGroupTask
+
+
+@dataclass
+class AsrNemoInferenceStage(ProcessingStage[FileGroupTask, SpeechEntry]):
+    """Stage that reads video files from storage and extracts metadata."""
+
+    model_name: str
+    filepath_key: str = "audio_filepath"
+    text_key: str = "text"
+    _name: str = "audio_inference"
+
+    def setup(self) -> None:
+        """Initialise heavy object self.asr_model: nemo_asr.models.ASRModel"""
+        self.asr_model = nemo_asr.models.ASRModel.from_pretrained(model_name=self.model_name)
+
+    def inputs(self) -> tuple[list[str], list[str]]:
+        """Define the input attributes required by this stage.
+
+        Returns:
+            Tuple of (top_level_attrs, data_attrs) where:
+            - top_level_attrs: ["data"] - requires VideoTask.data to be populated
+        """
+        return ["data"], []
+
+    def outputs(self) -> tuple[list[str], list[str]]:
+        """Define the output attributes produced by this stage.
+
+        Returns:
+            Tuple of (top_level_attrs, data_attrs) where:
+            - top_level_attrs: ["data"] - populates VideoTask.data
+            - data_attrs: ["text"] - predicted text
+        """
+        return ["data"], ["text"]
+
+    def transcribe(self, files: list[str]) -> list[str]:
+        """Run inference for speech recognition model
+         Args:
+            files: list of audio file paths.
+
+        Returns:
+            list of predicted texts.
+        """
+        outputs = self.asr_model.transcribe(files)
+        return [output.text for output in outputs]
+
+    def process_batch(self, tasks: list[FileGroupTask]) -> list[SpeechEntry]:
+        """Process a audio task by reading file bytes and extracting metadata.
+
+
+        Args:
+            task: VideoTask containing a Video object with input_video path set.
+
+        Returns:
+            The same VideoTask with video.source_bytes and video.metadata populated.
+            If errors occur, the task is returned with error information stored.
+        """
+        files = [task.data[0] for task in tasks]
+        outputs = self.transcribe(files)
+
+        audio_tasks = []
+        for i in range(len(outputs)):
+            text = outputs[i]
+            file_path = files[i]
+
+            entry = {self.filepath_key: file_path, self.filepath_key: text}
+
+            audio_task = SpeechEntry(
+                task_id=f"{file_path}_task_id",
+                dataset_name=f"{self.model_name}_inference",
+                data=entry,
+            )
```

Comment:
I think other params like `_stage_perf` and `_metadata` could be initialized here too.

Reply:
Yes, we initialize them here too since SpeechObject inherited from Task class

Reply:
Got it, keeping as default for now.